src/s25update.cpp

```diff
@@ -15,21 +15,25 @@
 // You should have received a copy of the GNU General Public License
 // along with Return To The Roots. If not, see <http://www.gnu.org/licenses/>.
 
+#ifdef _WIN32
```

Comment:
it's coming from curl.h it seems the new curl version does not include windows.h, or some kind of windows header chaos, since if curl is included after "s25update.h" it will fail with "SOCKET undefined blabla"

Reply:
Something about the lean and mean define maybe? Can I have a look at the error and the file (curl.h?) involved?

Reply:
```
In file included from /usr/src/cross-compiler/s25client-b/s25update/src/s25update.cpp:26:0:
/usr/x86_64-w64-mingw32/include/curl/curl.h:127:9: error: 'SOCKET' does not name a type
 typedef SOCKET curl_socket_t;
```

If I include "winsock2.h" before including curl, it works. yes that could be the WIN32_LEAN_AND_MEAN define.

seems to be a common "bug" with curl.